Cython/Compiler/ExprNodes.py

```diff
@@ -5034,13 +5037,17 @@ def coerce_to(self, dst_type, env):
         return super(SliceIndexNode, self).coerce_to(dst_type, env)
 
     def generate_result_code(self, code):
+        base_result = self.base.result()
+        result = self.result()
+
         if not self.type.is_pyobject:
-            error(self.pos,
-                  "Slicing is not currently supported for '%s'." % self.type)
+            if self.base.type.is_array:
+                code.putln("%s = %s;" % (result, base_result))
```

Comment:
If do you mean 
1) first to convert whole c-array to python list (typle, frozenset) ...
2) ... and then to use the new, created python list for creation of slice
ok, I will change !

Reply:
Hmm, no, I meant that the `SliceIndexNode` can do the copying of the array slice part itself, rather than having a separate node do it instead. It's the purpose of the `SliceIndexNode` to do the slicing, whatever the result is. And if the result is calculated by a copy function, so be it. But it's still up to the `SliceIndexNode` to call that function, right here in this place.

Reply:
ok, then clarified )